vtysh/vtysh_main.c

```diff
@@ -569,22 +577,30 @@ int main(int argc, char **argv, char **env)
 	/*
 	 * Setup history file for use by both -c and regular input
 	 * If we can't find the home directory, then don't store
-	 * the history information
+	 * the history information.
+	 * VTYSH_HISTFILE is prefered over command line
+	 * argument (-H/--histfile).
 	 */
-	homedir = vtysh_get_home();
-	if (homedir) {
-		snprintf(history_file, sizeof(history_file), "%s/.history_frr",
-			 homedir);
-		if (read_history(history_file) != 0) {
-			int fp;
-
-			fp = open(history_file, O_CREAT | O_EXCL,
-				  S_IRUSR | S_IWUSR);
-			if (fp != -1)
-				close(fp);
-
-			read_history(history_file);
-		}
+	if (getenv("VTYSH_HISTFILE")) {
+		snprintf(history_file, sizeof(history_file),
+			 getenv("VTYSH_HISTFILE"));
+	} else if (histfile) {
+		snprintf(history_file, sizeof(history_file), histfile);
+	} else {
+		homedir = vtysh_get_home();
+		if (homedir)
+			snprintf(history_file, sizeof(history_file),
+				 "%s/.history_frr", homedir);
+	}
+
+	if (read_history(history_file) != 0) {
+		int fp;
+
+		fp = open(history_file, O_CREAT | O_EXCL, S_IRUSR | S_IWUSR);
```

Comment:
`history_file` will be NULL here if there's no homedir

Reply:
NULL handled here properly. P.S. It was the same before.

Reply:
I mixed up `histfile` and and `history_file`, so yes, it won't be NULL.
But still, it's not as it was before. `history_file` will be uninitialized if there's no homedir. Previously, `read_history` was called only if there's homedir.

Reply:
I'll fix this.

Reply:
Fixed.